src/GSF.CarbonIntensity/test/ForecastHandlerTests.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using CarbonAware.Aggregators.CarbonAware;
+using CarbonAware.Tools.WattTimeClient;
+using GSF.CarbonIntensity.Exceptions;
+using GSF.CarbonIntensity.Handlers;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using static CarbonAware.Aggregators.CarbonAware.CarbonAwareParameters;
+
+namespace GSF.CarbonIntensity.Tests;
+
+[TestFixture]
+public class ForecastHandlerTests
+{
+    #pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+    private Mock<ILogger<ForecastHandler>> Logger { get; set; }
+    #pragma warning restore CS8618
+
+    [SetUp]
+    public void SetUp()
+    {
+        Logger = new Mock<ILogger<ForecastHandler>>();
+    }
+
+    [TestCase("Sydney", "eastus", "2022-03-07T01:00:00", "2022-03-07T03:30:00", 5, TestName = "All fields provided")]
```

Comment:
NIT: TestName is all that shows up in test runners.  Please include more context

EG `GetCurrentAsync calls aggregator: all fields`